state/remote/manta.go

```diff
@@ -80,3 +78,47 @@ func (c *MantaClient) Put(data []byte) error {
 func (c *MantaClient) Delete() error {
 	return c.Client.DeleteObject(c.Path, c.ObjectName)
 }
+
+func getCredentialsFromEnvironment() (cred *joyentauth.Credentials, err error) {
+
+	user := os.Getenv("MANTA_USER")
+	keyId := os.Getenv("MANTA_KEY_ID")
+	url := os.Getenv("MANTA_URL")
+	keyMaterial := os.Getenv("MANTA_KEY_MATERIAL")
+
+	if _, err := os.Stat(keyMaterial); err == nil {
+		// key material is a file path; try to read it
+		keyBytes, err := ioutil.ReadFile(keyMaterial)
+		if err != nil {
+			return nil, fmt.Errorf("Error reading key material from %s: %s",
+				keyMaterial, err)
+		} else {
+			block, _ := pem.Decode(keyBytes)
```

Comment:
This was cribbed directly from https://github.com/mitchellh/packer/blob/master/common/ssh/key.go#L27